charts/opentelemetry-demo/templates/jaeger.yaml

```diff
@@ -44,7 +44,7 @@ spec:
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ include "otel-demo.name" . }}-{{ .name }}
+  name: opentelemetry-demo-jaeger
```

Comment:
example of hard coding

Reply:
Yeah, we shouldn't do that. There must be a way. Let me think about it

Reply:
I've come up with a couple intrusive ideas, but I don't like the idea of change the collector chart specifically for this use case either.

Reply:
But maybe there is a more generic "oh you're using the collector as a subchart, then here is a solution to this problem" option.  We know there are other users using the collector as a subchart today.

Reply:
I think `tpl` could be a solution.

Reply:
I think it's a good idea. We should clearly define which values options support templating, not just jaeger endpoint as one workaround. support of value templating can be useful for other use cases

Reply:
Probably even all of them, if it's not going to add a lot of complexity

Reply:
@dmitryax I've pushed a version with tpl.  If we like it, I'll open a separate PR to get that into the collector chart.

Reply:
https://github.com/open-telemetry/opentelemetry-helm-charts/pull/395